addons/payment/models/payment_transaction.py

```diff
@@ -724,6 +746,8 @@ def classify_by_state(transactions_):
                      transactions `to process`, `processed`, and `in wrong state`.
             :rtype: tuple(recordset)
             """
+            if extra_allowed_states:
+                allowed_states.extend(extra_allowed_states)
```

Comment:
Do you mean that it's better to leave allowed_states as tuple, and instead of modifying that just create new list to avoid bugs connected to mutating the allowed_states?